tests/EachPromiseTest.php

```diff
@@ -342,4 +342,27 @@ public function testMutexPreventsGeneratorRecursion()
         $each->promise()->wait();
         $this->assertCount(20, $results);
     }
+
+    public function testIteratorWithSameKey()
+    {
+        $iter = function () {
+            yield 'foo' => 1;
+            yield 'foo' => 2;
+            yield 1 => 3;
+            yield 1 => 4;
+        };
+        $called = 0;
+        $each = new EachPromise($iter(), [
+            'fulfilled' => function ($value, $idx, Promise $aggregate) use (&$called) {
```

Comment:
Just for clarity: what's `idx` here is `key` above, right? (Sounds a bit confusing)

Reply:
Yes, I just kept the naming in the test file.